contracts/WildcardsQV.sol

```diff
@@ -114,32 +142,61 @@ contract WildcardsQV is Initializable {
     /////// vote function  ///////////
     ///////////////////////////////////
-    function vote(uint256 proposalIdToVoteFor, uint256 amount, uint256 sqrt) external {
+    function vote(uint256 proposalIdToVoteFor, uint256 amount, uint256 sqrt)
+        external
+    {
         // Check they are a wildcards user
-        require(wildCardToken.balanceOf(msg.sender)>0, "Does not own a WildCard");
+        require(
+            wildCardToken.balanceOf(msg.sender) > 0,
+            "Does not own a WildCard"
+        );
         // Check they have at least 1 wildcards loyalty token:
         require(amount >= 10**18, " Minimum vote one token");
-        // Check they are voting for a valid proposal: 
-        require(state[proposalIdToVoteFor] == ProposalState.Active, "Proposal not Active");
+        // Check they are voting for a valid proposal:
+        require(
+            state[proposalIdToVoteFor] == ProposalState.Active,
+            "Proposal not Active"
+        );
         // Check that they haven't yet voted for the proposal in this iteration
-        require(!hasUserVotedForProposalIteration[proposalIteration][msg.sender][proposalIdToVoteFor], "Already voted on this proposal");
+        require(
+            !hasUserVotedForProposalIteration[proposalIteration][msg
+                .sender][proposalIdToVoteFor],
+            "Already voted on this proposal"
+        );
         // Send their wildcards tokens to the burn address
-        require(loyaltyToken.transferFrom(msg.sender,burnAddress,amount), "Loyalty Token transfer failed");
+        require(
+            loyaltyToken.transferFrom(msg.sender, burnAddress, amount),
+            "Loyalty Token transfer failed"
+        );
         // Validate the square root
         require(sqrt.mul(sqrt) == amount, "Square root incorrect");
+
+        hasUserVotedForProposalIteration[proposalIteration][msg
+            .sender][proposalIdToVoteFor] = true;
+
         // Add the tokens to the total tally
         proposalVotes[proposalIteration][proposalIdToVoteFor] = proposalVotes[proposalIteration][proposalIdToVoteFor]
             .add(sqrt);
-        uint256 _currentVoteCount = proposalVotes[proposalIteration][proposalIdToVoteFor];
+
+
+            uint256 _currentVoteCount
+         = proposalVotes[proposalIteration][proposalIdToVoteFor];
         // Take note that the user has voted on this proposal so they can't do it again
-        hasUserVotedForProposalIteration[proposalIteration][msg.sender][proposalIdToVoteFor] = true;
+
         // Update currentWinner, currentHighestVoteCount, totalVotes
         if (_currentVoteCount > currentHighestVoteCount) {
             currentHighestVoteCount = _currentVoteCount;
             currentWinner = proposalIdToVoteFor;
         }
-        totalVotes = totalVotes.add(_currentVoteCount);
-        emit LogVote(proposalIdToVoteFor,sqrt,_currentVoteCount,totalVotes,msg.sender);
+
+        totalVotes = totalVotes.add(sqrt);
```

Comment:
This was adding _currentVoteCount, all the votes for the current proposal, instead of just sqrt, the votes being added from a result of this vote.